src/main/java/junmin/netflix/costreportservice/service/EPCostReportService.java

```diff
@@ -0,0 +1,169 @@
+package junmin.netflix.costreportservice.service;
+
+import junmin.netflix.costreportservice.exception.ErrorEnum;
+import junmin.netflix.costreportservice.exception.InvalidProductionException;
+import junmin.netflix.costreportservice.exception.InvalidReportException;
+//import junmin.netflix.costreportservice.model.EPItemizedCostEntity;
+import junmin.netflix.costreportservice.model.EPTotalCostEntity;
+import junmin.netflix.costreportservice.pojo.EPCostRecord;
+import junmin.netflix.costreportservice.repository.EPItemizedCostRepository;
+import junmin.netflix.costreportservice.repository.EPTotalCostRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.*;
+
+@Service
+public class EPCostReportService {
+	private EPTotalCostRepository epTotalCostRepo;
+	private EPItemizedCostRepository epItemizedCostRepo;
+
+	public EPCostReportService(EPTotalCostRepository epTotalCostRepo, EPItemizedCostRepository epItemizedCostRepo){
+		this.epTotalCostRepo = epTotalCostRepo;
+		this.epItemizedCostRepo = epItemizedCostRepo;
+	}
+
+	private final Logger LOGGER = LoggerFactory.getLogger(EPCostReportService.class);
+	private final String AMORIZEDCODE = "001";
+
+
+	public List<EPCostRecord> processEPCostReport(String prodName, List<EPCostRecord> costs) throws InvalidReportException
```

Comment:
I can invoke this method in JUnit with null value for prodName:
	@Test
	void processEPCostReportWithEmptyProdName() throws InvalidReportException {
		List<EPCostRecord> result = service.processEPCostReport(null, sampleMMEPCostInput());
	}

However, you had a good point here, while I'm validating for the sake of it, but in this case, since it's triggered by service controller call, the HttpServletRequest will bound back 404 if I'm not giving a prodName in the URL. So the empty prodName won't reach to this level. 

If there is no other places calling this method, it's safe to remove the check.